sdk/nodejs/runtime/settings.ts

```diff
@@ -242,14 +242,27 @@ export async function awaitFeatureSupport(): Promise<void> {
     const monitorRef = getMonitor();
     if (monitorRef !== undefined) {
         const store = getStore();
-        store.supportsSecrets = await monitorSupportsFeature(monitorRef, "secrets");
-        store.supportsResourceReferences = await monitorSupportsFeature(monitorRef, "resourceReferences");
-        store.supportsOutputValues = await monitorSupportsFeature(monitorRef, "outputValues");
-        store.supportsDeletedWith = await monitorSupportsFeature(monitorRef, "deletedWith");
-        store.supportsAliasSpecs = await monitorSupportsFeature(monitorRef, "aliasSpecs");
-        store.supportsTransforms = await monitorSupportsFeature(monitorRef, "transforms");
-        store.supportsInvokeTransforms = await monitorSupportsFeature(monitorRef, "invokeTransforms");
-        store.supportsParameterization = await monitorSupportsFeature(monitorRef, "parameterization");
+        const supports = await Promise.all(
+            [
+                "secrets",
+                "resourceReferences",
+                "outputValues",
+                "deletedWith",
+                "aliasSpecs",
+                "transforms",
+                "invokeTransforms",
+                "parameterization",
+            ].map((feature) => monitorSupportsFeature(monitorRef, feature)),
+        );
+
+        store.supportsSecrets = supports[0];
```

Comment:
It runs the promises in parallel, which seemed like a freebee to me since they don't need to be serialised. It's a bit noisier but we could avoid the indices with something like:

```
const [
  supportsSecrets,
  supportsResourceReferences,
  ...
] = await Promise.all([
  ...
].map(f => ...))

store.supportsSecrets = supportsSecrets
```

Which is a bit easier to guard against typos etc.

Reply:
Yeah, doing something like that would seem a lot less brittle to me.

Reply:
It _should_ be slightly faster as we fire off all the rpc requests at once rather than one at a time. But yes the indices is a bit sad. I did wonder if there was some clever dynamically setting fields by name code but that started looking even more complex.